### ISSUE TITLE
Not receiving password reset emails from firebase account that uses Sign In With Apple

### ISSUE THREAD

**Issue by yusuftor** [firebase/firebase-ios-sdk#6091]
### [REQUIRED] Step 1: Describe your environment
  * Firebase Component: Auth

### [REQUIRED] Step 2: Describe the problem

I successfully authenticate with Sign In With Apple using the iOS SDK and I hide my email address from Firebase by using Apple's private email relay service. I then try to send a password reset from the firebase console (or from the app itself), and I never receive the password reset email. 

I am using a custom domain, which I've added as an email source under "Configure Sign in with Apple for Email Communication" on the Apple Developer website. It's SPF verified. I've also tried adding the email itself noreply@mydomain.com. I've also tried adding the firebase email noreply@my-project.firebaseapp.com. However, I still don't receive a password reset email. Is it because Apple IDs don't have "passwords" that can be manually changed by a user, or am I doing something wrong? How can I test that emails work correctly?

**Comment by morganchen12** [comment]
Are you able to receive password reset emails without issue on a plain email+password account? Also, when you added an email to the Sign in with Apple account, did you also add a password?

**Comment by yusuftor** [comment]
Yes I am able to receive password resets for email+password. No I didn’t add a password, but the console does give me the option to send a password reset email, and the api does give a success code when entering my private email supplied by Apple.

**Comment by rosalyntan** [comment]
Thanks for filing this issue! I've filed b/162265835 for internal tracking.

**Comment by paulb777** [comment]
Closing for staleness. Please comment or open a new issue to reopen.

**Comment by oberbeck** [comment]
This is still reproducible today, in September 2025.

**Comment by ncooke3** [comment]
Hi all, what is the intended use case here? Password reset emails are intended for accounts with a password stored by Firebase. For accounts that are authorized by providers like Sign in with Apple, Sign in with Google, Facebook, etc., there is no Firebase-managed password to reset. The only exception here is when an account backed by one of the above providers is *linked* with the Email/Password provider, in which case the password reset email delivers to the email associated with the Email/Password linking.

To summarize:
1. Account 1 (Sign in with Apple only): there is no password that Firebase can reset so no email is delivered.
2. Account 2 (Sign in with Apple + Email/Password): a linked account that does have a Firebase-managed password so the email is delivered.
    - The important caveat here is that the password reset email needs to be sent to the email associated with the Email/Password portion of the account, not the email mapping to the iCloud account for SiwA. This only matters if the email used for SiwA is different than the email used for the linked Email/Password portion of the account.

Closing because this appears to be working as intended. Happy to reopen if I'm not addressing the same use case.
